Content.Shared/Chemistry/Solution/Solution.cs

```diff
@@ -43,6 +42,47 @@ public class Solution : IEnumerable<Solution.ReagentQuantity>, ISerializationHoo
         /// </summary>
         public Solution() { }
 
+        #region SolutionContainer Fields
+        /// <summary>
+        ///     If reactions will be checked for when adding reagents to the container.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("canReact")]
+        public bool CanReact { get; set; } = true;
+
+        /// <summary>
+        ///     Volume needed to fill this container.
+        /// </summary>
+        [ViewVariables]
+        public ReagentUnit EmptyVolume => MaxVolume - CurrentVolume;
+
+        public ReagentUnit RefillSpaceAvailable => EmptyVolume;
+        public ReagentUnit InjectSpaceAvailable => EmptyVolume;
+        public ReagentUnit DrawAvailable => CurrentVolume;
+        public ReagentUnit DrainAvailable => CurrentVolume;
```

Comment:
These should be partials or something and not on the default Solution.cs. Ideally Solution would be a struct (or close to it) and be general purpose.